### ISSUE TITLE
Subheadings do not account for combined Types (type + tags)

### ISSUE THREAD

**Issue by dschibster** [valentine195/obsidian-5e-statblocks#18]
Just noticed this when opening the statblock for Tarrasque, which has 

```json
{
"type" : {
"type" : "monstrosity",
"tags" : ["Titan"]
}
}
```

As its type, and in the beta it's parsed as [Object object], since in the Svelte file for the Subheading, properties are currently treated like Strings. I have tried out a quick solution for this, but I'm unable to compile it as I'm not too well-versed in TypeScript.

Could this be a solution?

```typescript
if(monster[property]['tags'] !== undefined){
 subheading.push(`${monster['type']}`); 
 subheading.push('('+`${monster['tags']}`.join(', ')+')');
}
else{
 subheading.push(`${monster[property]}`);
}
```

granted it's pretty specific, but my current assumption is this will only be relevant for this very case.

**Comment by valentine195** [comment]
Interesting find. Let me think about it. Maybe I need to parse properties recursively to combine things 🤔

**Comment by dschibster** [comment]
That would of course be the more generalized solution. Going through each property in an organized manner.

**Comment by dschibster** [comment]
Just noticed that there are a few other cases (not in the Subheading though, but actually in other descriptions) that there are sometimes objects being fed into the JSONs I receive from 5etools. Sometimes I see things like "bludgeoning, piercing, slashing damage from nonmagical sources" or things like AC having a case of normal and Mage Armor enabled AC in there.

Last thing I noticed, but I think that may just be inconsistent source material, is something I saw once or twice, is the sheer number for AC being in an Array and therefore also being recognized as a non-integer in parsing.

**Comment by valentine195** [comment]
Yeah, 5eTools is kind of a pain to deal with. I’ll make it generalized soon

**Comment by dschibster** [comment]
I figure! I wrote a quick parser for the source files, but I do **not** want to deal with cloned Sheets + Adjustments, like "this creature is an Animated Armor but with the following changes". Sheesh. :)

**Comment by valentine195** [comment]
This should be closed in 2.2.1
